interfaces/builtin/docker_support.go

```diff
@@ -657,7 +893,82 @@ ptrace (read, trace) peer=unconfined,
 
 # When kubernetes drives docker/containerd, it creates and runs files in the
 # container at arbitrary locations (eg, via pivot_root).
-/** rwlix,
+# Allow any file except for executing /snap/{snapd,core}/*/usr/lib/snapd/snap-confine
+# because in devmode confinement we will have a separate "x" transition on exec
+# rule that is in the policy that will overlap and thus conflict with this rule.
+# TODO: When we drop support for executing other snaps from devmode snaps (or 
+# when the AppArmor parser bugs are fixed) this can go back to the much simpler
+# rule:
+# /** rwlix,
+# but until then we need this set of rules to avoid exec transition conflicts.
+# See also the comment above the "change_profile unsafe /** -> docker-default," 
+# rule for more context.
+/[^s]** rwlix,
+/s[^n]** rwlix,
+/sn[^a]** rwlix,
+/sna[^p]** rwlix,
+/snap/{s[^n],c[^o]}** rwlix,
```

Comment:
As above - do we need a line above this of the form:

```
/snap/{[^s],[^c]}**,
```

Reply:
done